Vagga config

anxolerd opened the pull request:
# Vagga config
## Summary

Keep working on vagga config
## Changelog
- Added new container for running tests
- Added command for running nosetests 
- Better run command description
- Do not use pidfile when running app in development mode
- Decouple vagga from makefile. In fututre it would be great to get rid of that scary makefile.

anxolerd committed:
Keep working on vagga.yaml

- Make separate containers for running app and running tests
- Add command to run nosetests

Note:
  As `make test-nose` tries to install dependencies (once again,
  yeah), it is not suitale for running tests

  We need to get rid of Makefile cause vagga will do all the things we
  need

anxolerd committed:
Get rid of make commands in vagga config

I suppose vagga will be a great replacement for a makefile in the
fututre, so it would be great to make it independednt of Makefile so we
could easily drop the latter when it is time

anxolerd committed:
Do not use pidfile when running in dev mode

Why would we need a pidfile when running app in container in development
mode?

- Remove pidfile spec when running app in dev mode
- Remove unnecessary volume for pidfile

anxolerd committed:
Fix command name according to conventions

See http://vagga.readthedocs.io/en/stable/conventions.html#cmd-test

webknjaz commented on vagga.yaml:
```diff
@@ -22,22 +17,46 @@ containers:
-    env: !Command
-    container: gdg_dev
-    run: |
-      make run-dev
```
Bad idea. You'd have to support separate documentation/process explaining how to run everything on Mac environments.

webknjaz commented on vagga.yaml:
```diff
@@ -22,22 +17,46 @@ containers:
+      - !Py3Requirements requirements/test-env.txt
+    environ:
+      BLUEBERRYPY_CONFIG: "{}"
+      NOSE_TESTCONFIG_AUTOLOAD_YAML: "config/test/app.yml"
```
You don't have to wrap a normal string with quotes

webknjaz commented on vagga.yaml:
```diff
@@ -22,22 +17,46 @@ containers:
-        mode: 0o1777
+
+  gdg-test:
+    setup:
```
I'd add `dependencies: true` here as well

anxolerd committed:
Update to ubuntu xenial

Ubuntu xenial has python3.5 as default python3 interpreter. Using it
allows to avoid pain with adding custom repository.

Also I've dropped redundant packages and moved bower instalation to the
app-devel container. The first allows to keep container cleaner and the
second serves the purpose to not have node packages where they are not
needed

anxolerd committed:
Added simple vagrant configuration

This runs vagga inside a vagrant virtualbox ubuntu container.
There is still an issue with running a database which should be resolved

anxolerd committed:
Add vagrant to gitignore

$subj && edit comment for vagga

webknjaz commented on Vagrantfile:
```diff
@@ -0,0 +1,35 @@
+
+$script = <<SCRIPT
+VAGGA_CONF_FILE='/home/vagrant/.vagga/settings.yaml'
+if ! grep -q "ubuntu-mirror: " $VAGGA_CONF_FILE
```
Why do you do this?

anxolerd replied:
https://github.com/tailhook/vagga/issues/263

webknjaz replied:
Please state minimum explanation needed as a comment here with links provided.
The reason should be explicitly clear and obvious from the text.

webknjaz commented on Vagrantfile:
```diff
@@ -0,0 +1,35 @@
+# vi: set ft=ruby ts=2 sw=2 tw=2 et :
+
+
+$script = <<SCRIPT
```
?
